### PR TITLE
Integrate eager mode source code into onnxruntime repo

### PR DESCRIPTION
**Pull request by souptc**
**Description**: This PR is to integrate the ort eager mode code base into onnxruntime repo, including build and test.

**Motivation and Context**
- After team discussion, we decide to move eager mode codebase into onnxruntime codebase. Currently we put it under training folder as it is most frequently used with training, since it depends on pytorch.

### CODE REVIEW

**Review by pranavsharma — COMMENTED** [review]
1. What's the rationale for moving the eager mode code to the ORT repo? Is it because the training code is going to use this?
2. How many devs are working on eager mode? Will this trigger all our pipelines even if they're not exercising this code?

**Comment by souptc** [comment]
> 1. What's the rationale for moving the eager mode code to the ORT repo? Is it because the training code is going to use this?
There is a long discussion thread in the training group, basically the reason is we need to integrate with training code, while it is not ready to move the ORTModule code into pytorch/ort repo. And we also don't have enough permission on pytorch/ort repo, so the final decision is move the eager mode development into ort repo.

> 2. How many devs are working on eager mode? Will this trigger all our pipelines even if they're not exercising this code?
Around 3 to 6. I will sync with Changming to see how to filter the CI pipelines.

File: cmake/CMakeLists.txt [hunk 1]
```diff
@@ -156,6 +156,9 @@ option(onnxruntime_BUILD_OPSCHEMA_LIB "Build op schema library" ON)
 # option to enable custom operators in onnxruntime-extensions
 option(onnxruntime_ENABLE_EXTENSION_CUSTOM_OPS "Enable custom operators in onnxruntime-extensions" OFF)
 
+# pre-build python path
+option(prebuilt_PYTORCH_PATH "Path to pytorch installation dir")
```
[the remark this answers is not in the document]
Comment (souptc) [reply cmake/CMakeLists.txt]: resolved.

**Review by pranavsharma — DISMISSED** [review]
Spoke with @snnn offline and it looks like (1) we cannot filter pipelines based on a folder and (2) we've enough CI capacity and hence we don't need to worry about the impact of running unnecessary pipelines.

**Review by snnn — APPROVED** [review]

**Review by SherlockNoMad — APPROVED** [review]

**Review by SherlockNoMad — APPROVED** [review]

File: orttraining/orttraining/eager/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# ONNX Runtime Eager Mode Support for PyTorch
+
+## Build Instrcutions
+
+* Build Pytorch with commit: 0834a368181d18c8fd2429614fafca50fb412ce1
+* Install the pytorch build to your current environment
+* Build onnxruntime, make sure you enable training and python build together with eager mode:
+
+```bash
+./build.sh --enable_training --enalbe_pybind --build_eager_mode
```
Comment (pranavsharma) [inline orttraining/orttraining/eager/README.md:10]: typo: enable_pybind

**Pull request merged**